src/addon/validation.cpp

```diff
@@ -30,6 +30,16 @@ namespace {
 		"other", ""
 	};
 
+	// Reserved DOS device names on Windows XP and later.
+	const std::set<std::string> dos_device_names = {
+		"NUL", "CON", "AUX", "PRN",
```

Comment:
CLOCK$
COM5 ... COM9
LPT5 ... LPT9

Reply:
CLOCK$ is not exposed to Windows applications. Is it possible for more than 4 serial or parallel ports to be enumerated?

Reply:
Dunno about this millennium, but onceuponatime I had an 8-port serial card in a DOS box so it used to be so.

Frankly, all this seems way too close to the edge of the Rabbit Hole, and I'd suggest stopping before you eat any more of the Tea Cakes. There are times when it's better to ask the user which foot than to try to stop them shooting.

Reply:
You make it sound like this is about protecting Windows users from themselves, when in reality it's about protecting them from non-Windows users.